docs/pages/api-docs/data-grid/grid-api.md

```diff
@@ -10,85 +10,83 @@ import { GridApi } from '@mui/x-data-grid-pro';
 
```

Comment:
Switch to riche comparison to have something readable
I don't know why Git is marking every line as modified

Their are two changes

- Removal of `components` and `componentProps` keys (use `useGridRootProps` instead)
- Order change in type disjunction for two callbacks, I don't know why but it's not a real change